script.js

```diff
@@ -17,4 +18,270 @@ menuList.forEach((list) => {
+}
+
+function fillModal(project) {
+  const projectTechnologies = project.technologies.split(',');
+  const modalContainer = document.querySelector('.project-details');
+  modalContainer.innerHTML = '';
+  const modal = document.createElement('div');
+  modal.classList.add('modal');
+
+  const header = document.createElement('header');
+  header.classList.add('modal-header');
+  header.innerHTML = `
+  <h2>${project.name}</h2>
+  <img alt="x button to close the modal" class='cancel-modal' src="images/cancel.png">
+  `;
+  modal.appendChild(header);
+  const techList = document.createElement('ul');
+  techList.classList.add('project-tech');
+  techList.classList.add('tech-details');
+
+  projectTechnologies.forEach((tech) => {
+    const list = document.createElement('li');
+    list.innerHTML = `<li><a href="#">${tech}</a></li>`;
+    techList.appendChild(list);
+  });
+  modal.appendChild(techList);
+  const projectImage = document.createElement('div');
+  const flexContainer = document.createElement('div');
+  flexContainer.classList.add('flex-container');
+  modal.appendChild(flexContainer);
+  projectImage.classList.add('project-img');
+  projectImage.innerHTML = `<img alt="image illustrating project" src="${project.image}">`;
+
+  flexContainer.appendChild(projectImage);
+  const projectDescription = document.createElement('div');
+  projectDescription.classList.add('pro-description');
+  projectDescription.innerHTML = `
+      <p class="descript">${project.description}</p>
+      <div class="pro-links">
+          <a href="${project.liveLink}" class="project-button about-btn">See Live</a>
+          <a href="${project.sourceLink}" class="project-button about-btn">See Source</a>
+      </div>
+     `;
+  flexContainer.appendChild(projectDescription);
+  modalContainer.appendChild(modal);
+}
+
+window.addEventListener('load', () => fillPage());
+
+const modalContainer = document.querySelector('.project-details');
+
+projectSection.addEventListener('click', (e) => {
+  if (e.target.classList.contains('project-button')) {
+    modalContainer.classList.remove('invisible');
+    const name = e.target.id;
+    const project = getProjectByName(name);
+    fillModal(project);
+  }
+});
+
+modalContainer.addEventListener('click', (e) => {
+  if (e.target.classList.contains('cancel-modal')) {
+    modalContainer.classList.add('invisible');
+  }
+});
+function displayError(error) {
+  const errorDiv = document.querySelector('.error-div');
+  const errorMessageDiv = document.querySelector('.error-message');
+  errorDiv.classList.remove('invisible');
+  errorMessageDiv.textContent = error;
+}
+
+const cancelMessage = document.querySelector('#cancel-message');
+cancelMessage.addEventListener('click', () => {
+  const errorDiv = document.querySelector('.error-div');
+  errorDiv.classList.add('invisible');
+});
+
+const contactForm = document.querySelector('#cont-form');
+contactForm.addEventListener('submit', (e) => {
+  e.preventDefault();
```

Comment:
- Please check the form validation if it works properly on your branch `feature-form-validation` because it did not validate the email address when I entered all capital letters, as seen in the video below. 

##### Testing your project contact form


https://user-images.githubusercontent.com/58771507/174292898-d05c54c1-bdf2-48fe-ab07-8480beadaed0.mp4